### PR TITLE
Add events functionality to the website.

### PR DESCRIPTION
**Pull request by jwage**
Building a tool to allow us to host events on the website with the ability to charge participants via Stripe.

## Crons

Run this every minute to download event participants from Stripe checkouts to save them to the database and e-mail the participant their ticket.

```bash
./doctrine event-participants --save --email
```

## Events

screencapture-lcl-doctrine-project-org-events-html-2019-05-11-19_08_48

## View Event

screencapture-lcl-doctrine-project-org-events-doctrine-online-conference-2019-html-2019-05-11-21_27_32

## Stripe Checkout

screencapture-checkout-stripe-pay-ppage-1EZCFuAbh7z6kx2ce2RF6w7D-2019-05-12-02_38_04

## Checkout Success

Stripe sends the participant to this screen after successfully checking out.

checkoutsuccess

## Event Participants

Since we don't have a backend I am using the Stripe API to pull a list of participants for an event. This is so that we can do things like invite them to the Zoom meeting, e-mail them, etc.

```
$ ./doctrine event-participants doctrine-online-conference-2019

Doctrine Online Conference 2019 Participants
============================================

 ---------------------------- ----------
  E-Mail                       Quantity
 ---------------------------- ----------
  jwage@doctrine-project.org   1
  jonwage@gmail.com            5
 ---------------------------- ----------
```

## Event Participant E-Mail

We will send this e-mail to participants 7 days before the event begins.

Screen Shot 2019-05-12 at 10 12 40 PM

## Event Finished

Zoom sends the participant to this screen after they exit the webinar.

eventfinished

### CODE REVIEW

**Comment by SenseException** [comment]
A Build was successful and I could use the new event sites (minus the Stripe checkout).

**Comment by jwage** [comment]
You can use a test card 4242-4242-4242-4242 in non prod environments.

**Comment by SenseException** [comment]
That PR got too big. Looking at the latest commit, I can't tell which code I already reviewed and which changes are new. Redoing this review will take hours of my time again. Please use multiple commits for your PRs and squash them later. Maybe I'll find enough time this weekend.

One test failed and Scrutinizer met a failure condition.

**Comment by jwage** [comment]
Sorry, I normally do that but someone else asked me to squash them eagerly in another PR review so I started doing it that way.

File: lib/Model/EventType.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?php
```
[the remark this answers is not in the document]
Comment (jwage) [reply lib/Model/EventType.php]: This is the way we've started doing it in other Doctrine libraries so I was following that convention.

**Comment by jwage** [comment]
@SenseException 

> You have a location information in the participant email template, but I miss them on the events list and in the event page itself.

The location information is not shown on the list currently but it is on the event page and in the e-mail template. https://github.com/doctrine/doctrine-website/pull/306/files#diff-3ea24881d9bf3223378097608a3059a0R79

> A user specific time-zone would be a nice to have for the events. This can be interesting if the event is a webinar where people around the world can participate.

I agree that would be cool. It would require some frontend javascript work.

> Tests are failing

Fixed the failing test in the latest commit.

**Pull request merged**